tests/nightly/Jenkinsfile

```diff
@@ -0,0 +1,194 @@
+      sleep 2
+    }
+  }
+}
+
+def docker_run(platform, function_name, use_nvidia, shared_mem = '500m') {
+  def command = "ci/build.py --docker-registry ${env.DOCKER_CACHE_REGISTRY} %USE_NVIDIA% --platform %PLATFORM% --shm-size %SHARED_MEM% /work/runtime_functions.sh %FUNCTION_NAME%"
+  command = command.replaceAll('%USE_NVIDIA%', use_nvidia ? '--nvidiadocker' : '')
+  command = command.replaceAll('%PLATFORM%', platform)
+  command = command.replaceAll('%FUNCTION_NAME%', function_name)
+  command = command.replaceAll('%SHARED_MEM%', shared_mem)
+
+  sh command
+}
+
+try {
+  stage('NightlyTests'){
+    parallel 'RATCheck: CPU': {
+      node('mxnetlinux-cpu') {
+        ws('workspace/nt-RATTest') {
+          init_git()
+          docker_run('ubuntu_nightly_cpu', 'nightly_test_rat_check', false)
+        }
+      }
+    },
+    'CompilationWarnings: GPU': {
+      node('mxnetlinux-gpu') {
+        ws('workspace/nt-compilationTest') {
+          init_git()
+          docker_run('ubuntu_nightly_gpu', 'nightly_test_compilation_warning', false)
+        }
+      }
+    },
+    'InstallationGuide: CPU': {
+      node('mxnetlinux-gpu') {
+        ws('workspace/nt-Installation-cpu') {
+          init_git()
+          docker_run('ubuntu_base_cpu', 'nightly_test_installation ubuntu_python_cpu_virtualenv', false)
+          docker_run('ubuntu_base_cpu', 'nightly_test_installation ubuntu_python_cpu_pip', false)
+          //Docker Install Test is currently disabled and tracked here: https://github.com/apache/incubator-mxnet/issues/11288
+          //docker_run('ubuntu_base_cpu', 'nightly_test_installation ubuntu_python_cpu_docker', false)
+          docker_run('ubuntu_base_gpu', 'nightly_test_installation ubuntu_python_cpu_source', false)
+        }
+      }
+    },
+    'InstallationGuide: GPU': {
+      node('mxnetlinux-gpu') {
+        ws('workspace/nt-Installation-gpu') {
+          init_git()
+          docker_run('ubuntu_base_gpu', 'nightly_test_installation ubuntu_python_gpu_virtualenv', true)
+          docker_run('ubuntu_base_gpu', 'nightly_test_installation ubuntu_python_gpu_pip', true)
+          ///Docker Install Test is currently disabled and tracked here: https://github.com/apache/incubator-mxnet/issues/11288
+          //docker_run('ubuntu_base_gpu', 'nightly_test_installation ubuntu_python_gpu_docker', true)
+          docker_run('ubuntu_base_gpu', 'nightly_test_installation ubuntu_python_gpu_source', true)
+        }
+      }
+    },
+    'PipTest: GPU': {
+      node('mxnetlinux-gpu') {
+        ws('workspace/nt-pipTest') {
+          init_git()
+        }
+      }
+    },
+    'Amalgamation-atlas: CPU': {
+      node('mxnetlinux-cpu') {
+        ws('workspace/nt-amalgamation1') {
+          init_git()
+          //sh "ci/build.py --platform ubuntu_nightly_cpu /work/runtime_functions.sh nightly_test_amalgamation USE_BLAS=atlas"
+          docker_run('ubuntu_nightly_cpu', 'nightly_test_amalgamation USE_BLAS=atlas', false)
```

Comment:
As we discussed, I did this to parallelize amalgamation. 
each of these takes approx 20mins to run and hence running in parallel makes sense since currently the pipeline completes in about 27mins.

Reply:
Ok